dev/Repeater/ItemsRepeater.idl

```diff
@@ -442,178 +510,219 @@ unsealed runtimeclass StackLayout : VirtualizingLayout
+     }
+
+     [MUX_PREVIEW]
+     [webhosthidden]
+     unsealed runtimeclass SelectionModel : Windows.UI.Xaml.Data.INotifyPropertyChanged
+     {
+         SelectionModel();
+
+         event Windows.Foundation.TypedEventHandler<SelectionModel, SelectionModelSelectionChangedEventArgs> SelectionChanged;
+         event Windows.Foundation.TypedEventHandler<SelectionModel, SelectionModelChildrenRequestedEventArgs> ChildrenRequested;
+         Object Source { get; set; };
+         Boolean SingleSelect { get; set; };
+         IndexPath AnchorIndex { get; set; };
+         IndexPath SelectedIndex { get; set; };
+         Object SelectedItem { get; };
+         Windows.Foundation.Collections.IVectorView<Object> SelectedItems { get; };
+         Windows.Foundation.Collections.IVectorView<IndexPath> SelectedIndices { get; };
+         [method_name("SetAnchorIndex")]
+         void SetAnchorIndex(Int32 index);
+         [method_name("SetAnchorIndexWithGroup")]
+         void SetAnchorIndex(Int32 groupIndex, Int32 itemIndex);
+         [default_overload] [method_name("Select")]
+         void Select(Int32 index);
+         [method_name("SelectWithGroup")]
+         void Select(Int32 groupIndex, Int32 itemIndex);
+         void SelectAt(IndexPath index);
+         [default_overload] [method_name("Deselect")]
+         void Deselect(Int32 index);
+         [method_name("DeselectWithGroup")]
+         void Deselect(Int32 groupIndex, Int32 itemIndex);
+         void DeselectAt(IndexPath index);
+         [default_overload] [method_name("IsSelected")]
+         Windows.Foundation.IReference<Boolean> IsSelected(Int32 index);
+         [method_name("IsSelectedWithGroup")]
+         Windows.Foundation.IReference<Boolean> IsSelected(Int32 groupIndex, Int32 itemIndex);
+         Windows.Foundation.IReference<Boolean> IsSelectedAt(IndexPath index);
+         [default_overload] [method_name("SelectRangeFromAnchor")]
+         void SelectRangeFromAnchor(Int32 index);
+         [method_name("SelectRangeFromAnchorWithGroup")]
+         void SelectRangeFromAnchor(Int32 groupIndex, Int32 itemIndex);
+         void SelectRangeFromAnchorTo(IndexPath index);
+         [default_overload] [method_name("DeselectRangeFromAnchor")]
+         void DeselectRangeFromAnchor(Int32 index);
+         [method_name("DeselectRangeFromAnchorWithGroup")]
+         void DeselectRangeFromAnchor(Int32 groupIndex, Int32 itemIndex);
+         void DeselectRangeFromAnchorTo(IndexPath index);
+         void SelectRange(IndexPath start, IndexPath end);
+         void DeselectRange(IndexPath start, IndexPath end);
+         void SelectAll();
+         void ClearSelection();
+
+         protected void OnPropertyChanged(String propertyName);
+     }
+
+     [MUX_PREVIEW]
+     [webhosthidden]
+     [MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+     [MUX_PROPERTY_CHANGED_CALLBACK_METHODNAME("OnPropertyChanged")]
+     unsealed runtimeclass FlexboxLayout : NonVirtualizingLayout
```

Comment:
I'm not too familiar with idl, can these be put in a separate file so this isn't as massive and in the future it becomes clearer which things are changing specific layout implementation contracts vs. the base ItemsRepeater one?

Reply:
I totally agree with @michael-hawker , I think it would be better to split off a new (or several new) IDL files for layouts.